_posts/2020-02-20-community-build.md

```diff
@@ -0,0 +1,148 @@
+---
+layout: blog-detail
+post-type: blog
+by: Seth Tisue
+title: "Scala 2 community build achieves completion"
+---
+
+The Scala 2 community build is a large collection of open-source Scala
+code that we build and run every day in order to validate changes to
+the Scala compiler and standard library.
+
+If you're not familiar with the community build at all, this
+[introduction](https://www.scala-lang.org/2018/01/16/community-build-growth.html),
+from January 2018, still applies.
+
+A year ago, we [updated
+you](https://www.scala-lang.org/2019/01/18/community-build.html) on
+improvements made up through January 2019.  The post you are reading
+highlights the progress the community build has made since then.
+
+In short, we believe **the Scala 2 community build is now essentially
+complete**. Still, we will continue to run and maintain it throughout
+Scala 2's lifetime.
+
+We describe the build as complete because:
+
+* It has successfully supported the release of Scala 2.13
+  and the migration of the ecosystem to 2.13.
+* It is large and diverse enough to support future 2.13.x releases.
+* It is now entirely green on JDK 8 and 11 and nearly green on JDK 14.
+
+Details below.
+
+## Overall growth
+
+The 2.13 community build is now the main one.  (The 2.11 and 2.12
+builds still exist, but they have required only light maintenance
+for some time now.)
+
+The 2.13 build currently has **209 repos** and a total of **3.4 million
+lines** of code.  (A year ago, on 2.12, we had 185 repos and 3.2
+million lines.  On 2.13, we had only 79 repos.)  It takes about
```

Comment:
```suggestion
million lines.  On 2.12, we had only 79 repos.)  It takes about
```

off by one?

Reply:
No, it's correct. The 2.12 build was big, and the 2.13 build was a lot smaller. It wasn't until later in the year that the 2.13 build caught up, because we had to wait for maintainers to move to 2.13.

Maybe I should add ", as we waited for maintainers to add 2.13 support", to make the narrative clearer?

Reply:
Ah -- yes, I think some form of clarification would be useful.

Reply:
59f7e129bdfca3aca077d7bcbe5f6cc269d2715d